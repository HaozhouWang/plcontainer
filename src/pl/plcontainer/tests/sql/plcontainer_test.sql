-- first some tests of basic functionality
--
-- better succeed
--
select writeFile();
-- make sure the file doesn't exist on the host filesystem
\! ls -l /tmp/foo

select rlog100();
select pg_spi_exec('select 1');
select pyint(NULL::int2);
select pyint(1::int2);
select pyint(1::int4);
select pyint(1::int8);
select pyfloat(1.2::float4);
select pyfloat(1.2::float8);
select pybool('f');
select pybool('t');
select pylog100();
select pylog(10000, 10);
select concat(fname, lname) from users order by 1;
select concatall();
select nested_call_three('a');
select nested_call_two('a');
select nested_call_one('a');
select invalid_function();
select invalid_syntax();
select rlog100_shared();
select pylog100_shared();