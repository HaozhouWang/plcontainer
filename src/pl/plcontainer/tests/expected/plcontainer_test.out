-- first some tests of basic functionality
--
-- better succeed
--
select writeFile();
        writefile         
--------------------------
 wrote foobar to /tmp/foo
(1 row)

-- make sure the file doesn't exist on the host filesystem
\! ls -l /tmp/foo
ls: cannot access /tmp/foo: No such file or directory
select rlog100();
 rlog100 
---------
 2
(1 row)

select pg_spi_exec('select 1');
 pg_spi_exec 
-------------
 1
(1 row)

select pyint(NULL::int2);
 pyint 
-------
      
(1 row)

select pyint(1::int2);
 pyint 
-------
     1
(1 row)

select pyint(1::int4);
 pyint 
-------
     1
(1 row)

select pyint(1::int8);
 pyint 
-------
     1
(1 row)

select pyfloat(1.2::float4);
 pyfloat 
---------
     1.2
(1 row)

select pyfloat(1.2::float8);
 pyfloat 
---------
     1.2
(1 row)

select pybool('f');
 pybool 
--------
 f
(1 row)

select pybool('t');
 pybool 
--------
 t
(1 row)

select pylog100();
 pylog100 
----------
        2
(1 row)

select pylog(10000, 10);
 pylog 
-------
     4
(1 row)

select concat(fname, lname) from users order by 1;
  concat   
-----------
 janedoe
 johndoe
 ricksmith
 willemdoe
(4 rows)

select concatall();
       concatall       
-----------------------
 jane,john,rick,willem
(1 row)

select nested_call_three('a');
 nested_call_three 
-------------------
 a
(1 row)

select nested_call_two('a');
      nested_call_two       
----------------------------
 {'nested_call_three': 'a'}
(1 row)

select nested_call_one('a');
                  nested_call_one                  
---------------------------------------------------
 {'nested_call_two': "{'nested_call_three': 'a'}"}
(1 row)

select invalid_function();
ERROR:  exception occurred: (plcontainer.c:238)
DETAIL:  'module' object has no attribute 'foobar'
select invalid_syntax();
ERROR:  exception occurred: (plcontainer.c:238)
DETAIL:  ('invalid syntax', ('<string>', 7, 1, '\n'))
select rlog100_shared();
 rlog100_shared 
----------------
 2
(1 row)

select pylog100_shared();
 pylog100_shared 
-----------------
               2
(1 row)

